### ISSUE TITLE
Clean the VM

### ISSUE THREAD

**Issue by MarvinLavechin** [srvk/DiViMe#64]
Big ~~spring~~ fall clean !
What needs to be done : 
- [ ]  Merge **toolbo**x, **tools**, **varia** folders into **tools**
- [ ] Create **launcher** folder to store all of the scripts that launch something
- [ ] Create **data/temporary** folder where all the temporary data will be store (if asked by the user)

Don't hesitate if you have ideas to improve the way we're dealing with the files hierarchy.
It's now or never !

**Comment by riebling** [comment]
lots of tools break when they can't find things where they expect. I'm not sure it's fair to require tool creators (some no longer available, or some tools we may not have source code for) to make changes to their code to accomodate new folder names.

**Comment by riebling** [comment]
Config files normally found in /vagrant have gone missing in the reorganization. More than one tool uses the HTK feature config file `(ERROR) [1] in configManager : cFileConfigReader::openInput : cannot open input file '/vagrant/MED_2s_100ms_htk.conf'!`
Also .theanorc (which is in the reorganized repository) and also a whole folder of configuration files named https://github.com/srvk/DiViMe/tree/master/conf which exists "outside the VM" for a reason: it makes it easier for nonexperts to change a configuration local to their machine, but then tools in the VM reads the new configuration to work in a different way. 
`conf` is currently mostly for enhancements to the VM (the slurm batch queuing system for example) but I see @MarvinLavechin is using a new subfolder `conf/vad`. I propose that we preserve conf in the reorganization, and if we want to move forward with changing tool source code to work in new folder layours, we move the 2 or more tools that depend on MED_2s_100ms_htk.conf to get their configs from /vagrant/conf/ instead of just /vagrant, so that at least that ugly filename won't be there making the list of files look ugly :)

**Comment by alecristia** [comment]
> lots of tools break when they can't find things where they expect. I'm not sure it's fair to require tool creators (some no longer available, or some tools we may not have source code for) to make changes to their code to accomodate new folder names.

For the past tools, I propose we either fix it ourselves or remove the tools. We are not committed to preserving all tools that are there. The goal is not to preserve the VM as it is now, but rather to build a sensible, well-organized platform that others can build on and use easily. Perhaps the changes we suggested with Marvin were not ideal for this, so we are very open to discussion of how best to do this.

**Comment by alecristia** [comment]
> Config files normally found in /vagrant have gone missing in the reorganization. More than one tool uses the HTK feature config file `(ERROR) [1] in configManager : cFileConfigReader::openInput : cannot open input file '/vagrant/MED_2s_100ms_htk.conf'!`
> Also .theanorc (which _is_ in the reorganized repository) and also a whole folder of configuration files named https://github.com/srvk/DiViMe/tree/master/conf which exists "outside the VM" for a reason: it makes it easier for nonexperts to change a configuration local to their machine, but then tools in the VM read the new configurations to work in a different way.
> `conf` is currently mostly for enhancements to the VM (the slurm batch queuing system for example) but I see @MarvinLavechin is using a new subfolder `conf/vad`. I propose that we preserve conf in the reorganization, and if we want to move forward with changing tool source code to work in new folder layouts, we alter the source of the 2 or more tools (much as this terrifies me) that depend on MED_2s_100ms_htk.conf to get their configs from /vagrant/conf/ instead of just /vagrant, so that at least that ugly MED_2s_100ms_htk.conf filename won't be there making the list of files look ugly :)

I think part of this comments point out a bug I introduced -- sorry about that.

And part is about adding back a folder I did not know about. Of course, we can use it, in which case it would be ideal if there was some documentation explaining what is in there and why, so that in the future people can re-use it.

**Comment by riebling** [comment]
Super! Moving on... (I've been working from home this morning since waking up a couple hours ago)
I've now made enough changes to the new scripts in launcher/ (and possibly utils/) to get the tests to pass.
This does not mean that they all work in the desired way yet, simply that they WORK. If it's OK to commit all my changes, you could use this as a starting point for "something that works" then make incremental changes to finish making them work with the new reorganized folder paths.

In particular, things that I changed in order to make the tests all pass, that may need more changes to work to fully work in the reorganized way:

* Put a copy of Opensmile back in vagrant's home folder (tools that are hard coded to find it there) by hand
* Put OpenSAT back in the home folder (it has numerous hard coded dependencies that require it to be there) by hand
* put G/ back in the home folder (tools hard code that location) by hand
* Refactored and fixed bugs in numerous launcher/ and utils/ scripts relating to pathname confusions we've encountered: /vagrant vs. /home/vagrant and DATADIR being absolute path vs. relative (data vs /vagrant/data)
* This includes some cases of scripts that take as an argument a folder name (eg "data") intended to be in /vagrant (/vagrant/data) being actually passed full paths ("/vagrant/data") [ in the eval.sh stack of scripts' use of audio_dir]
* LOTS of fixes where more tool names needed to be changed from e.g: tool_sad or tool_diar to toolSad and toolDiar 

Things that still need work:
* eval.sh sub-scripts still use temp_ref and temp_sys rather than temp/temp_ref and temp/temp_sys (there were too many changes for me to make to get it to work the temp/temp_ref way)
* many places in scripts define paths with trailing slashes,  that then get concatenated into bigger pathnames that work fine, but if you print them out the paths look like e.g: /vagrant//data/VanDam-Public/BN32//folder-name - not a real bug :)
* G/ folder is back in /home/vagrant, again because tools have hard coded dependencies on it being there
* Changes to Vagrantfile to ensure these tools are back where they were -or- a larger effort of changing the tool sources (something I was hoping we could leave fixed) to work with new locations which involves:
  - putting modified tools in a new branch or release tag or repository
  - maintaining multiple versions of tool source code, if we wish to support people currently using them in current DiViMe deployments
  - possibly updating tools' internal documentation 
  - possibly updating system-level documentation

So I'm mostly writing to ask if it's OK to commit my changes to test.sh that "work", and the launcher/ scripts I've modified to work with it, in hopes that development of the reorganized branch can continue with now a way to test tools again, and starting from these "working" ones, then make further refinements to them use new temp folder layout.

**Comment by alecristia** [comment]
Eric and myself discussed having a launcher and utils folders as separate repos versus within the divime folder. We decided for the latter because:

- for someone who wants to install a new tool inside VM, this means forking ONE thing and not 2 or 3,
- for someone reviewing a pull request, this means reviewing changes in one repo, not 2-3
- it'll be easier for tool makers to edit launcher etc. from the host machine rather than only through the VM
- all the launch/util scripts are hidden neatly inside folders so the GitHub page for DiViMe home folder still looks clean enough

The only disadvantage is that this makes the divime repo heavier in size, but this does not seem to be an issue given that launchers and utils are all light files.

We also decide to add repos to PATH, which will allow very clean and easy to read code for calling tools: `vagrant ssh -c launchMe.sh data`

We also decide that launchers will be by default read-only, to avoid accidental editing

**Comment by alecristia** [comment]
- [ ] straightening out chaos
That's one box you should check!

**Comment by riebling** [comment]
one of the far above checkboxes goes away (compile selectively the docs based on tool failures)

**Comment by alecristia** [comment]
good point, that actually takes care of 2 problems - because now the online and local docs are the same. I'm adding a line to bootstraph.sh to build the local docs.

Also, I think the last unchecked box, "Inside each tool subfolder (e.g., tools/OpenSAT), there will be a wrapper that specifies the behavior to adopt regarding the files that need to be processed*" is no longer relevant, right? Since you're systematically having all the tools analyze file by file, so by the same token, this routine should break up long files into subfiles?
